rclpy/rclpy/node.py

```diff
@@ -185,6 +180,11 @@ def __init__(
             self._parameters.update(self._parameter_overrides)
             self._descriptors.update({p: ParameterDescriptor() for p in self._parameters})
 
+        # Clock that has support for ROS time.
+        self._clock = ROSClock()
+        self._time_source = TimeSource(node=self)
+        self._time_source.attach_clock(self._clock)
```

Comment:
@jubeira why the relocation? Can we add a comment explaining it?

Reply:
Yes; how about https://github.com/ros2/rclpy/pull/396/commits/7de9dd52e06c9f9b3d861f860a7cef87ed93d738?
The explanation is that `TimeSource` constructor now might declare `use_sim_time` and ends up calling `_set_parameters_atomically` method, which in turn requires the weakref and the [event publisher](https://github.com/ros2/rclpy/pull/396/files/4c46f6162287f5617691da33a7858aa6afe88650#diff-129fafcf34261ae2f52bd43718815308R169) to be initialized before.
Moving these lines here also ensures processing parameter overrides before creating the  source (which is needed in case `use_sim_time` is overridden to `True`).